OpenRA.Mods.Common/Graphics/DefaultSpriteSequence.cs

```diff
@@ -204,13 +204,13 @@ public DefaultSpriteSequence(ModData modData, TileSet tileSet, SpriteCache cache
 							"{0}: Sequence {1}.{2}: Length must be <= Frames.Length"
 							.F(info.Nodes[0].Location, sequence, animation));
 
-					if (Start < 0 || Start + (Facings - 1) * Stride + Length > frameCount)
+					if (Frames == null && (Start < 0 || Start + (Facings - 1) * Stride + Length > frameCount))
 						throw new InvalidOperationException(
 							"{5}: Sequence {0}.{1} uses frames [{2}..{3}], but only 0..{4} actually exist"
 							.F(sequence, animation, Start, Start + (Facings - 1) * Stride + Length - 1, frameCount - 1,
 								info.Nodes[0].Location));
 
-					if (ShadowStart + (Facings - 1) * Stride + Length > frameCount)
+					if (ShadowStart > 0 && ShadowStart + (Facings - 1) * Stride + Length > frameCount)
```

Comment:
Shouldn't this be `>= 0`? If there are no shadow frames the value should be `-1` (that is at least the default value).

Reply:
We don't have cases for `ShadowStart: 0` but I think they are valid so updated.